src/AlgebraicGeometry/Schemes/Varieties/Types.jl

```diff
@@ -0,0 +1,94 @@
+@doc Markdown.doc"""
+    AbsAffineAlgebraicSet <: AbsSpec
+
+An affine, geometrically reduced scheme of finite type over a field.
+"""
+abstract type AbsAffineAlgebraicSet{BaseField<:Field, RingType} <:AbsSpec{BaseField, RingType}# where {BaseField <:Field, RingType}
+end
+
+@doc Markdown.doc"""
+    AbsProjectiveAlgebraicSet <: AbsProjectiveScheme
+
+An affine, geometrically reduced scheme of finite type over a field.
+"""
+abstract type AbsProjectiveAlgebraicSet{BaseField<:Field, RingType} <:AbsProjectiveScheme{BaseField, RingType}# where {BaseField <:Field, RingType}
+end
+
+################################################################################
+#
+# Abstract types for varieties
+#
+################################################################################
+
+@doc Markdown.doc"""
+    AbsAffineVariety <: AbsAffineAlgebraicSet
+
+An affine, geometrically integral scheme of finite type over a field.
+"""
+abstract type AbsAffineVariety{BaseField<:Field, RingType} <:AbsAffineAlgebraicSet{BaseField, RingType}# where {BaseField <:Field, RingType}
+end
+
+@doc Markdown.doc"""
+    AbsProjectiveVariety <: AbsProjectiveAlgebraicSet
+
+A projective variety over a field.
+
+That is a projective, geometrically integral scheme over a field.
+"""
+abstract type AbsProjectiveVariety{BaseField<:Field, GradedRingType<:Ring} <: AbsProjectiveAlgebraicSet{BaseField, GradedRingType} end
+
+@doc Markdown.doc"""
+    AbsCoveredVariety <: Scheme
+
+A separated, geometrically integral scheme of finite type over a field.
+
+Note that we allow reducible varieties.
+"""
+abstract type AbsCoveredVariety{BaseField<:Field} <: AbsCoveredScheme{BaseField} end
+
+################################################################################
+#
+# Abstract types for curves
+#
+################################################################################
+
+@doc Markdown.doc"""
+    AbsAffineCurve <: AbsAffineVariety
+
+A curve in affine space.
+
+An affine curve is an affine variety of dimension one.
+"""
+abstract type AbsAffineCurve{BaseField<:Field, RingType<:Ring} <: AbsAffineVariety{BaseField, RingType} end
+
+@doc Markdown.doc"""
+    AbsProjectiveCurve <: AbsProjectiveVariety
+
+A projective curve embedded in an ambient projective space.
+"""
+abstract type AbsProjectiveCurve{BaseField<:Field, RingType<:Ring} <: AbsProjectiveVariety{BaseField, RingType} end
+
+
+@doc Markdown.doc"""
+    AbsCoveredCurve
+
+A curve represented in terms of a covering.
+
+A curve is a geometrically integral scheme of finite type over a field which is
+of dimension one.
```

Comment:
This sentence is somehow hard to read. Maybe
"A curve is a geometrically integral scheme of dimension 1 which is of finite type over a field"